### ISSUE TITLE
use the plugin in a standalone script

### ISSUE THREAD

**Issue by SIGISLV** [gacarrillor/InvisibleLayersAndGroups#3]
I try to use your class in standalone script but I don't have acess to the iface so I rewrite the __init__ so that it can instanciate the model and the treeview from the root of a project, but when I save the project Layers that I want them to be hide remain visible. This is because the TreeView that I have instanciate is differrent form the treeView of the iface and it's only in this treeview where my layers are invisible.

My question is there a way to synchronize my LayerTrreView and the project's TreeView ? so it that can be saved by the project ?

**Comment by gacarrillor** [comment]
Can you share your script to reproduce the issue?

**Comment by SIGISLV** [comment]
Ok as I was preparing the script for sharing it and I found out that my custom Qgis has made me joke. I open the project where I hidden node with my custom Qgis and the same project with a regular Qgis. My custom Qgis doesn't hide my layers whereas the other hide them correctly. I don't know why. Anyway here's the main part of the trick to hide node :

    from qgis.core import QgsProject, QgsLayerTreeModel
    from qgis.qui import QgsLayerTreeView
   
    project = QgsProject.instance()
    project.read("Path/of/the/project.qgs")
    root = project.layerTreeRoot()
    model = QgsLayerTreeModel(root)
    ltv = QgsLayerTreeView()
    ltv.setModel(model)

    layer = project.mapLayersByName('layer name you like')[0]
    node = root.findLayer( layer.id())

    index = model.node2index( node )
    # Here we hide the node
    ltv.setRowHidden( index.row(), index.parent(), True )
    # Here the property is written
    node.setCustomProperty( 'nodeHidden', 'true')
    # Here the property is save in the project
    ltv.setCurrentIndex(model.node2index(root))

Thank for the suggestion so that I could see my problem.

**Comment by gacarrillor** [comment]
Note that before reading any QGIS project, InvisibleLayersAndGroups sets a connection from `QgsProject.readProject` SIGNAL to `self.readHiddenNodes`(exactly [here](https://github.com/gacarrillor/InvisibleLayersAndGroups/blob/master/invisibleLayersAndGroups.py#L38)).

That's a must, because as soon as you load a project, the plugin takes care of hiding nodes based on the `nodeHidden` custom property they might have stored in the project.

Could be that the issue you're facing?

**Comment by SIGISLV** [comment]
Yeah you are right ! Do you interested about my standalone script to hide node ?

**Comment by gacarrillor** [comment]
You can post it so future readers know how you solved the issue.

Regards
